src/pages/storage/StoragePoolSelectTable.tsx

```diff
@@ -0,0 +1,143 @@
+import { FC } from "react";
+import { Button, MainTable } from "@canonical/react-components";
+import ScrollableTable from "components/ScrollableTable";
+import StoragePoolSize from "pages/storage/StoragePoolSize";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { fetchStoragePools } from "api/storage-pools";
+
+interface Props {
+  project: string;
+  onSelect: (pool: string) => void;
+  disablePool?: {
+    name: string;
+    reason: string;
+  };
+}
+
+const StoragePoolSelectTable: FC<Props> = ({
+  project,
+  onSelect,
+  disablePool,
+}) => {
+  const { data: pools = [], isLoading } = useQuery({
+    queryKey: [queryKeys.storage],
+    queryFn: () => fetchStoragePools(project),
+  });
+
+  const headers = [
+    { content: "Name", sortKey: "name" },
+    { content: "Driver", sortKey: "driver" },
+    { content: "Size", className: "size" },
+    { content: "Status", sortKey: "status" },
+    {
+      content: "Volumes",
+      sortKey: "volumes",
+      className: "u-align--right",
+    },
+    { "aria-label": "Actions", className: "actions" },
+  ];
+
+  const rows = pools.map((pool) => {
+    const volumes =
+      pool.used_by?.filter((url) => !url.startsWith("/1.0/profiles")) ?? [];
+    const totalVolumeCount = volumes.length;
+
+    const disableReason = disablePool?.name === pool.name && disablePool.reason;
```

Comment:
I think we should find a better way to visualize the disabled row. Maybe remove the hover effect for it, or we could try muted text in that row. Or hide it instead of disabling?

Reply:
going with muted text, show not-allowed cursor for a disabled row. wdyt?

Reply:
Sounds OK.
The not allowed cursor is barely used on the web, maybe just remove the hover effect and use the default pointer?